tests/unit/text.test.tsx

```diff
@@ -0,0 +1,60 @@
+import { configureStore, combineReducers } from "@reduxjs/toolkit";
+import React, { ReactNode } from "react";
+import { Provider } from "react-redux";
+import { expect, test } from "vitest";
+import { renderHook, act } from "@testing-library/react-hooks";
+import textReducer from "../../src/core/text.slice";
+import { useText } from "../../src/core/utils/text";
+
+const store = configureStore({
+  reducer: combineReducers({
+    text: textReducer
+  }),
+  preloadedState: {
+    text: {
+      data: {
+        simpleText: "This is a @simple test",
+        placeholdersText: `{"type":"simple","text":["This is a text with a @placeholder embedded. Does it work? @result it does!"]}`,
+        pluralText: `{"type":"plural","text":["You have 1 material on the waiting list","You have @count materials on the waiting list"]}`
+      }
+    }
+  }
+});
+
+const Wrapper = ({ children }: { children: ReactNode }) => (
+  <Provider store={store}> {children} </Provider>
+);
+
+test("Should render expected text strings with placeholders and plural handling", () => {
```

Comment:
This test needs to be ran manually, right? because we don't have any setup to run it with Github actions.